modules/layers/src/icon-layer/icon-manager.js

```diff
@@ -304,29 +307,33 @@ export default class IconManager {
     }
   }
 
-  _loadIcons(icons) {
+  async _loadIcons(icons) {
     const ctx = this._canvas.getContext('2d');
 
-    for (const icon of icons) {
-      loadImage(icon.url).then(imageData => {
-        const id = getIconId(icon);
-        const {x, y, width, height} = this._mapping[id];
+    for (let i = 0; i < icons.length; i++) {
+      const imageData = await loadImage(icon.url);
 
-        const data = resizeImage(ctx, imageData, width, height);
+      if (i === icons.length - 1) {
```

Comment:
- This marks loaded when the last image is loaded, not when all images are loaded.
- What happens when the icon set changes?

Reply:
> * What happens when the icon set changes?

reset `loaded` to false when there are more icons required to be loaded.